h2o-py/h2o/utils/shared_utils.py

```diff
@@ -353,6 +358,144 @@ def slice_is_normalized(s):
+    :return: Successful path name or None if failed
+    """
+    for root, dirs, files in os.walk(path):
+        if name in files:
+            return os.path.join(root, name)
+
+def check_json(json_str):
+    try:
+        json_object = json.loads(json_str)
+    except ValueError:
+        raise RuntimeError("Error: Given JSON string does not look like valid JSON string.")
+    return True
+
+def predict_json(mojo_model, json, genmodelpath=None, labels=False, classpath=None, javaoptions='-Xmx4g', show_debug=False):
+    """
+    MOJO scoring function to take a pandas data frame as json string and use MOJO model as zip file to score
+
+    :param mojo_model: This is the MOJO model file name which is download after model was build using H2O, you can pass it two ways:
+           1. Full mojo model path i.e. /Users/avkashchauhan/src/github.com/h2oai/h2o-tutorials/tutorials/python_mojo_scoring/gbm_prostate_new.zip
+           2. mojo name first i.e. gbm_prostate_new.zip and genmodelpath = /Users/avkashchauhan/src/github.com/h2oai/h2o-tutorials/tutorials/python_mojo_scoring
+           3. only mojo model name: In this case system uses os.path.abspath to get the folder where mojo is and then look for h2o-genmodel.jar in the same location
+    :param json:  convert pandas frame to json (pd.to_json) and pass here
+    :param genmodelpath: This is the local file system folder name where both h2o-genmodel.jar and MOJO zip file is stored.
+    :param labels:     (Optional) True : Shows results, False: Does no show results and just pass result to given object
+    :param javaoptions: (Optional) These are the Java string options given by the user
+    :param show_debug: True/False - Default ( Use True to see the full Java command line)
+    :return: score as json values
+    """
+    # Verifying JSON
+    if show_debug:
+        print(json)
+
+    check_json(json)
+
+    # Checking java
+    java = H2OLocalServer._find_java()
+
+    jver_bytes = subprocess.check_output([java, "-version"], stderr=subprocess.STDOUT)
+    jver = jver_bytes.decode(encoding="utf-8", errors="ignore")
+    if show_debug:
+        print("  Java Version: " + jver.strip().replace("\n", "; "))
+    if "GNU libgcj" in jver:
+        raise H2OStartupError("Sorry, GNU Java is not supported for H2O.\n"
+                              "Please download the latest 64-bit Java SE JDK from Oracle.")
+    """
+    if "Client VM" in jver:
+        warn("  You have a 32-bit version of Java. H2O works best with 64-bit Java.\n"
+             "  Please download the latest 64-bit Java SE JDK from Oracle.\n")
+    # genmodelpath > must start and ends with "/" or add to it
+
+    if sys.platform == "win32":
+        separator = ";"
+    else:
+        separator = ":"
+    """
+
+    separator = os.pathsep
+    file_separator = os.sep
+
+    gen_model_arg = "." + separator
+
+    ## Working on mojo_model
+    mojo_model_path=None
+
+    head, tail = os.path.split(mojo_model)
```

Comment:
I commented in earlier stages of the PR on the logic of finding mojo and the jar file. Cannot find it anymore, copying from Slack message instead:

"michalk [9:23 AM] 
@avkash I posted my comments. Basically a lot of the code concerns finding mojos, constructing paths. I think the behavior should be simpler to avoid surprises to the user. Consider this case: the user downloads the mojo to his Downloads folder and forgets to specify the genmodel path. The current implementation finds recursively some genmodel.jar in the Downloads folder. If the user has multiple H2O version in the downloads folder, it will just pick one of them (probably the oldest but it is generally not predictable)."